b2luigi/batch/processes/gbasf2.py

```diff
@@ -0,0 +1,764 @@
+
+        gbasf2_input_dataset = get_setting("gbasf2_input_dataset", default=False, task=self.task)
+        if gbasf2_input_dataset is not False:
+            gbasf2_command_str += f" -i {gbasf2_input_dataset} "
+
+        gbasf2_n_repition_jobs = get_setting("gbasf2_n_repition_job", default=False, task=self.task)
+        if gbasf2_n_repition_jobs is not False:
+            gbasf2_command_str += f" --repetition {gbasf2_n_repition_jobs} "
+
+        # now add some additional optional options to the gbasf2 job submission string
+
+        # whether to ask user for confirmation before submitting job
+        force_submission = get_setting("gbasf2_force_submission", default=True, task=self.task)
+        if force_submission:
+            gbasf2_command_str += " --force "
+
+        # estimated cpu time per sub-job in minutes
+        cpu_minutes = get_setting("gbasf2_cputime", default=False, task=self.task)
+        if cpu_minutes is not False:
+            gbasf2_command_str += f" --cputime {cpu_minutes} "
+
+        # estimated number or processed events per second
+        evtpersec = get_setting("gbasf2_evtpersec", default=False, task=self.task)
+        if evtpersec is not False:
+            gbasf2_command_str += f" --evtpersec {evtpersec} "
+
+        # gbasf2 job priority
+        priority = get_setting("gbasf2_priority", default=False, task=self.task)
+        if priority is not False:
+            assert 0 <= priority <= 10, "Priority should be integer between 0 and 10."
+            gbasf2_command_str += f" --priority {priority} "
+
+        # gbasf2 job type (e.g. User, Production, ...)
+        jobtype = get_setting("gbasf2_jobtype", default=False, task=self.task)
+        if jobtype is not False:
+            gbasf2_command_str += f" --jobtype {jobtype} "
+
+        # additional basf2 options to use on grid
+        basf2opt = get_setting("gbasf2_basf2opt", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" --basf2opt='{basf2opt}' "
+
+        # optional string of additional parameters to append to gbasf2 command
+        gbasf2_additional_params = get_setting("gbasf2_additional_params", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" {gbasf2_additional_params} "
+
+        gbasf2_command = shlex.split(gbasf2_command_str)
+        return gbasf2_command
+
+    def _write_path_to_file(self):
+        """
+        Serialize and save the ``basf2.Path`` returned by ``self.task.create_path()`` to a python pickle file.
+        """
+        try:
+            path = self.task.create_path()
+        except AttributeError as err:
+            raise Exception(
+                "Gbasf2 batch process can only used with tasks that generate basf2 paths with "
+                "a ``create_path()`` method, e.g. are an instance of ``Basf2PathTask``."
+            ) from err
+
+        path.add_module("Progress")
```

Comment:
Please do not change the path of the user. I used to do this, but at some point users will have problems because the number of modules they expect is not the same.

Reply:
Removed in ea01ba1